File: RoomReservationBackend/Models/Room.cs
namespace RoomReservationBackend.Models
{
    public class Room
    {
        public int RoomId { get; set; }
        public string Name { get; set; } = string.Empty; // Changed to match your model

        public int Capacity { get; set; }
        public string Location { get; set; } = string.Empty;
        public string Amenities { get; set; } = string.Empty;

        // Navigation property - A room can have multiple reservations
        public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>(); // Add this
    }
}

File: room_reservation_app/lib/Models/room.dart
class Room {
  final int roomId;
  final String name;
  final int capacity;
  final String location;

  Room({
    required this.roomId,
    required this.name,
    required this.capacity,
    required this.location,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      roomId: json['roomId'],
      name: json['name'],
      capacity: json['capacity'],
      location: json['location'],
    );
  }
}
